matrix_client/api.py

```diff
@@ -523,13 +523,18 @@ def _send(self, method, path, content=None, query_params={}, headers={},
 
         response = None
         while True:
-            response = requests.request(
-                method, endpoint,
-                params=query_params,
-                data=content,
-                headers=headers,
-                verify=self.validate_cert
-            )
+            try:
+                response = requests.request(
+                    method, endpoint,
+                    params=query_params,
+                    data=content,
+                    headers=headers,
+                    verify=self.validate_cert
+                )
+            except requests.exceptions.RequestException as e:
+                raise MatrixHttpLibError(
+                    "Something went wrong in {} requesting {}".format(method, endpoint), e
```

Comment:
Personal preference is to pass in method and endpoint as parameters to the exception and format it inside the constructor, in case I might want to disassemble it as a handler down the line.